inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java

```diff
@@ -302,12 +315,43 @@ private Label createNoRecommendationLabel() {
             @Override
             public boolean isVisible()
```

Comment:
Do not overwrite `isVisible`. Instead implement `onConfigure()`:

```
                @Override
                protected void onConfigure()
                {
                    super.onConfigure();
                    setVisible(startLearning && !hasUnseenRecommendation && !hasSkippedRecommendation);
                }
```

We even have a `LambdaBehavior` now which could be used without even creating an anonymous subclass:

```
Label noRecommendation = new Label("noRecommendationLabel", "There are no further suggestions.")
  .add(LambdaBehavior.onConfigure(_this -> _this.setVisible(startLearning && !hasUnseenRecommendation && !hasSkippedRecommendation));
```

When a component can become invisible, it is usually a good idea to have it output a placeholder, otherwise its visibility status cannot be changed via AJAX:

```
Label noRecommendation = new Label("noRecommendationLabel", "There are no further suggestions.")
  .add(LambdaBehavior.onConfigure(_this -> _this.setVisible(startLearning && !hasUnseenRecommendation && !hasSkippedRecommendation));
noRecommendation.setOutputMarkupPlaceholderTag(true);
```

Same applies to other places where isVisible() is overwritten.

Reply:
Groovy also uses "it". I'm a bit torn between generic names like "_this" or "it" and using a more speaking name like "component".

Reply:
changed